### PR TITLE
Minor fixes to the source installation guide

### PR DESCRIPTION
**Pull request by Saransh-cpp**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all`
- [ ] The documentation builds: `$ python run-tests.py --doctest`

You can run unit and doctests together at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by agriyakhetarpal — APPROVED** [review]
Looks good, thanks! It's just missing the command that describes how to run the `integration` tests (maybe we should keep `nox -s integration` and `nox -s tests` documented too, inclusive of your changes).

File: docs/source/user_guide/installation/install-from-source.rst [hunk 1]
```diff
@@ -162,22 +162,30 @@ You can use Nox to run the unit tests and example notebooks in isolated virtual
...
-.. tab:: GNU/Linux, MacOS & Windows
-
-	.. code:: bash
+.. code:: bash
 
-		nox -s tests
+	nox
 
-will run the full test suite (integration and unit tests).
+will run pre-commit, install ``Linux`` dependencies, and run the unit tests.
```
Comment (arjxn-py) [inline docs/source/user_guide/installation/install-from-source.rst:169]:
```suggestion
will run pre-commit, install ``Linux`` dependencies, and run tests.
```
As suggested in [this comment](https://github.com/pybamm-team/PyBaMM/pull/3084#discussion_r1253653304) we can run all tests by default.

File: docs/source/user_guide/installation/install-from-source.rst [hunk 2]
```diff
@@ -162,22 +162,30 @@ You can use Nox to run the unit tests and example notebooks in isolated virtual
...
 This can take several minutes.
 
-It is often sufficient to run the unit tests only. To do so, use
+To just run the unit tests, use
```
Comment (arjxn-py) [inline docs/source/user_guide/installation/install-from-source.rst:172]:
```suggestion
To run unit tests, use
```

**Review by arjxn-py — CHANGES_REQUESTED** [review]
Thanks @Saransh-cpp, Looks great.
Made some suggestions, noxfile could also be changed accordingly.

File: docs/source/user_guide/installation/install-from-source.rst (hunk 1 above)
Comment (Saransh-cpp) [reply docs/source/user_guide/installation/install-from-source.rst:169]: I think you misread my comment. I suggested moving the docs around, nothing in particular with running integration tests as the default session.

**Review by arjxn-py — APPROVED** [review]

File: docs/source/user_guide/installation/install-from-source.rst [hunk 3]
```diff
@@ -127,33 +125,35 @@ You can now activate the environment with
...
 
-If you intend to contribute to the development of PyBaMM, it is convenient to install in "editable mode", along with useful tools for development and documentation:
+If you intend to contribute to the development of PyBaMM, it is convenient to
+install in "editable mode", along with all the optional dependencies and useful
+tools for development and documentation:
 
 .. code:: bash
 
-	  pip install -e .[dev,docs]
+	  pip install -e .[all,dev,docs]
```
Comment (tinosulzer) [inline docs/source/user_guide/installation/install-from-source.rst:147]: doesn't "all" include dev and docs?

Comment (tinosulzer) [reply docs/source/user_guide/installation/install-from-source.rst:147]: got it, makes sense

**Review by tinosulzer — APPROVED** [review]

**Pull request merged**